cost/aws/elb/clb_unused/AWS_Delete_Unused_CLB.pt

```diff
@@ -0,0 +1,242 @@
+    query "Version", "2012-06-01"
+    query "LoadBalancerNames.member.1",val(iter_item, "loadBalancerName")
+    header "Accept", "application/json"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "DescribeTagsResponse.DescribeTagsResult.TagDescriptions[*]") do
+    field "tags" do
+    collect jmes_path(col_item,"Tags") do
+      field "tagKey", jmes_path(col_item,"Key")
+      field "tagValue", jmes_path(col_item,"Value")
+    end
+    end
+    field "loadBalancerName", val(iter_item, "loadBalancerName")
+    field "availabilityZones", val(iter_item, "availabilityZones")
+    field "instancesList", val(iter_item, "instancesList")
+   end
+  end
+end
+
+#https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_DescribeInstanceHealth.html
+datasource "ds_aws_instances_health" do
+iterate $ds_aws_clb_list
+  request do
+  auth $aws
+    host "elasticloadbalancing.amazonaws.com"
+    path "/"
+    verb "GET"
+    query "Action", "DescribeInstanceHealth"
+    query "Version", "2012-06-01"
+    query "LoadBalancerName",val(iter_item, "loadBalancerName")
+    header "Accept", "application/json"
+  end
+  result do
+  encoding "json"
+    collect jmes_path(response, "DescribeInstanceHealthResponse.DescribeInstanceHealthResult.InstanceStates[*]") do
+    field "description", jmes_path(col_item,"Description")
+    field "state", jmes_path(col_item,"State")
+    field "instanceId", jmes_path(col_item,"InstanceId")
+    field "loadBalancerName", val(iter_item, "loadBalancerName")
+    end
+  end
+end
+
+#https://www.cloudconformity.com/conformity-rules/ELB/unused-elastic-load-balancers.html
+script "js_aws_clb_filter_map", type: "javascript" do
+  parameters "ds_aws_clb_list_tags", "ds_aws_instances_health", "param_exclude_tags"
+
+  result "content"
+  code <<-EOS
+  var content=[]
+  for(var i=0; i<ds_aws_clb_list_tags.length; i++){
+    clb=ds_aws_clb_list_tags[i]
+
+    // Constructing availabilityZones with comma separated to display in detail_template
+    var availabilityZones=clb['availabilityZones']
+    var s_availabilityZones=""
+    for(var j=0; j < availabilityZones.length; j++){
+      availabilityZone = availabilityZones[j]
+      s_availabilityZones = s_availabilityZones + ', '+ availabilityZone
+    }
+
+    // Constructing IntanceIds with comma separated to display in detail_template
+    var instancesList = clb['instancesList']
+    var s_instanceIds=""
+    for(var j=0; j < instancesList.length; j++){
+      instanceId = instancesList[j]
+      s_instanceIds = s_instanceIds + ', '+ instanceId
+    }
+
+    //Constructing Tags of individual CLB into key=value format. Check, if the tag present in entered param_exclude_tags, ignore the CLB if the tag matches/present.
+    var tags = clb['tags']
+    var isTagMatched=false
+    var tagKeyValue=""
+    for(var j=0; j < tags.length; j++){
+      tag = tags[j]
+      if(param_exclude_tags.indexOf(tag['tagKey']+'='+tag['tagValue']) !== -1){
```

Comment:
Since you are comparing strings you should take case sensitivity into consideration.

Reply:
Handled case sensitive for tags comparison.

Reply:
@avinashbhashyam-rs this needs to be resolved also